test/JDBC/expected/check_constraints_definition.out

```diff
@@ -0,0 +1,213 @@
+	c_binary binary(8) check(c_binary > cast(0xfe as binary(8))),
+	c_varbinary varbinary(8) check(c_varbinary > cast(0xfe as varbinary(8)))
+);
+go
+
+-- float(7) will be displayed as real in the constraint definition  
+create table test_tsql_cast(
+	c_float float check(c_float < cast(133.230182309832423 as float) and c_float < cast(133.230182309832423 as float(7))),
+	c_real real check(c_real < cast(133.230182309832423 as real)),
+	c_decimal decimal check(c_decimal < cast(cast($133.23 as money) as decimal) and c_decimal < cast(133.230182309832423 as decimal(5,3))),
+	c_numeric numeric check(c_numeric < cast(133.230182309832423 as numeric) and c_numeric < cast(133.230182309832423 as numeric(5,3))),
+);
+go
+
+create table test_tsql_collate(
+	c_varchar varchar check(c_varchar <> cast('sflkjasdlkfjf' as varchar(12)) COLLATE bbf_unicode_cp1_ci_as),
+	c_char char check(c_char <> cast('sflkjasdlkfjf' as char(7)) COLLATE bbf_unicode_cp1_ci_as),
+	c_char1 char check(c_char <> cast('abcd' as char(7)) COLLATE "default"),
+	c_nchar nchar check(cast(c_nchar as nchar(7)) <> cast('sflkjasdlkfjf' as nchar(7)) COLLATE bbf_unicode_cp1_ci_as),
+);
+go
+
+create table test_datetime(
+	c_time time check(cast(c_time as pg_catalog.time) < cast('09:00:00' as time) and c_time < cast('09:00:00' as time(6))),
+	c_date date check(c_date < cast('2001-01-01' as date)),
+	c_datetime datetime check(c_datetime < cast('2020-10-20 09:00:00' as datetime)),
+	c_datetime2 datetime2 check(c_datetime2 < cast('2020-10-20 09:00:00' as datetime2) and c_datetime2 < cast('2020-10-20 09:00:00' as datetime2(6)) ),
+	c_datetimeoffset datetimeoffset check(c_datetimeoffset < cast('12-10-25 12:32:10 +01:00' as sys.datetimeoffset) and c_datetimeoffset < cast('12-10-25 12:32:10 +01:00' as datetimeoffset(4))),
+	c_smalldatetime smalldatetime check(c_smalldatetime < cast('2007-05-08 12:35:29.123' AS smalldatetime)),
+);
+go
+
+create table test_functioncall(
+	col1 sys.nvarchar,
+	check (isjson(col1) > 0),
+	check (right(col1,1) <> ','),
+	check (ltrim(col1) <> ''),
+	check (CONVERT([nvarchar](128),(sys.getutcdate() AT TIME ZONE [col1]))<>'')
+);
+go
+
+create schema sch1;
+create type sch1.udd_float from sys.float;
+create type sch1.udd_datetime from sys.datetime;
+create type sch1.udd_time from time(6);
+go
+
+-- Inside check constraint definition alias datatypes are not allowed
+create table test_udd(
+	c_udd sch1.udd_float check(c_udd < 133.230182309832423),
+	c_udd1 sch1.udd_datetime check(c_udd1 < '2020-10-20 09:00:00'),
+	c_udd2 sch1.udd_time check(c_udd2 < '09:00:00'),
+);
+go
+
+create table test_null(a int, b int, check(a IS NOT NULL), CONSTRAINT constraint1 check (a>10));
+go
+
+create table test_upper(a char, check (upper(a) in ('A','B')));
+go
+
+alter table test_null add constraint constraint2 check(a<=20 and b>a and (a between 40 and 50));
+go
+
+set ansi_nulls on;
+go
+create table test_null1(a int, b int, check(a <> NULL), check(b = NULL));
+go
+	
+set ansi_nulls off;
```

Comment:
I think by default ansi_nulls is on so we should reset it to default after this test